api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingBatchUpdateService.java

```diff
@@ -38,38 +39,60 @@ public class FreeTierBillingBatchUpdateService {
 
   private final BigQueryService bigQueryService;
 
+  private final GoogleProjectPerCostDao googleProjectPerCostDao;
+
+  private final WorkspaceDao workspaceDao;
+
   private static final int MIN_USERS_BATCH = 5;
   private static final int MAX_USERS_BATCH = 999;
   public static final Range<Integer> batchSizeRange =
       Range.closed(MIN_USERS_BATCH, MAX_USERS_BATCH);
 
   @Autowired
   public FreeTierBillingBatchUpdateService(
+      GoogleProjectPerCostDao googleProjectPerCostDao,
       UserDao userDao,
       FreeTierBillingService freeTierBillingService,
       Provider<WorkbenchConfig> workbenchConfigProvider,
+      WorkspaceDao workspaceDao,
       BigQueryService bigQueryService) {
+    this.googleProjectPerCostDao = googleProjectPerCostDao;
     this.userDao = userDao;
     this.freeTierBillingService = freeTierBillingService;
     this.workbenchConfigProvider = workbenchConfigProvider;
     this.bigQueryService = bigQueryService;
+    this.workspaceDao = workspaceDao;
   }
 
   /**
    * Takes in List of : user and all BQ cost of workspace for the user
    *
-   * @param userBQCostReq
+   * @param userIdList
    */
-  public void checkAndAlertFreeTierBillingUsage(List<UserBQCost> userBQCostReq) {
+  public void checkAndAlertFreeTierBillingUsage(List<Long> userIdList) {
     Map<String, Double> userWorkspaceBQCosts = new HashMap<String, Double>();
 
+    Set<String> googleProjectsForUserSet =
+        userIdList.stream()
+            .map(
+                (userId) -> {
+                  return workspaceDao.getGoogleProjectForUser(userId);
```

Comment:
this line can be more compact as well
.map(workspaceDao::getGoogleProjectForUser)

Reply:
There is no need to query them one by one. We should add a new method
like 
```
  Set<String> getGoogleProjectByCreatorIn(List<Long> creators);
```

(And I think we might not need to write custom query if we use `creator` since it matches column name).